core/unit_test/TestDeepCopy.hpp

```diff
@@ -211,4 +211,41 @@ TEST(TEST_CATEGORY, deep_copy_alignment) {
   }
 }
 
+namespace Impl {
+template <typename T, typename SHMEMTYPE>
+using ShMemView =
+    Kokkos::View<T, Kokkos::LayoutRight, SHMEMTYPE, Kokkos::MemoryUnmanaged>;
+}
+
+TEST(TEST_CATEGORY, deep_copy_scratch) {
+  using TestDeviceTeamPolicy =
+      Kokkos::TeamPolicy<Kokkos::DefaultExecutionSpace>;
+  using ShmemType = Kokkos::DefaultExecutionSpace::scratch_memory_space;
+
+  const int bytes_per_team = 256, bytes_per_thread = 256;
+  const int threads_per_team = 1;
+
+  TestDeviceTeamPolicy policy(1, threads_per_team);
+  auto team_exec = policy.set_scratch_size(1, Kokkos::PerTeam(bytes_per_team),
+                                           Kokkos::PerThread(bytes_per_thread));
+
+  const int N = 8;
+
+  Kokkos::parallel_for(
+      team_exec,
+      KOKKOS_LAMBDA(
```

Comment:
Lambda won't work with NVCC because gtest wraps the function in a class in the body of a private member function.  Also this would need to be within a preprocessor directive that checks whether lambda are enabled :(